main.cc

```diff
@@ -64,5 +64,13 @@ int main() {
   readTrain(&trainData);
   readTest(&testData);
   train(batch, layers, trainData, testData, lr, momentum, weightDecay);
+
+  vector<Sample> kaggleData;
```

Comment:
Please put this out of `int main()`. The stack size is much smaller than the heap size. The `kaggleData` will be pretty large.